### ISSUE TITLE
Replace `write_to_png` and `write_to_svg` by `write`

### ISSUE THREAD

**Issue by MaximeBouton** [sisl/AutoViz.jl#42]

**Comment by mattuntergassmair** [comment]
I think the way we have currently solved this is not sufficient, as it allows syntax such as

```
write("stadium.svg", c) # all good
write("stadium.png", c) # saves with .png suffix, but is actually an svg
```
We should really implement this in a way that the format of the file is deduced from the filename suffix (I think the plotting package does something very similar)

**Comment by MaximeBouton** [comment]
One of the challenge is that depending on the type of `c` we would not be able to save in any format, but I guess we could return an error message in that case. 
- Any `CairoSurface` can be saved as `.png` 
- For saving to `svg` or `pdf` it would need the correct surface type. If it is not the correct type, the write function could return an error message that tells how to fix it.

**Comment by MaximeBouton** [comment]
I will try to implement that soon, let's wait for this feature before merging #43

**Comment by mattuntergassmair** [comment]
Cool, sounds good. Yeah I'd say erroring is fine in that case and we can always leave `write_to_svg` etc. around and then just call that function from within `Base.write()` for convenience.

**Comment by MaximeBouton** [comment]
Distinguishing between `svg` and `pdf` is not trivial as `CairoPDFSurface` and `CairoSVGSurface` are actually of the same type (`CairoSurfaceIOStream{UInt32}`).

**Comment by MaximeBouton** [comment]
I think it should be solved in #43 now, related issue JuliaGraphics/Cairo.jl#316
